progect-management-app/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,238 @@
+import { Component, OnInit, Output, EventEmitter, ChangeDetectorRef} from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import {DashboardService, Column, Task} from "./services/dashboard.service";
+import { MatDialog } from '@angular/material/dialog';
+import {DialogComponent} from "../dialog/dialog.component";
+import { Location } from '@angular/common';
+import {ColumnFormComponent} from "./column-form/column-form.component";
+import {first, map} from "rxjs";
+import {TaskFormComponent} from "./task-form/task-form.component";
+import {TaskEditFormComponent} from "./task-edit-form/task-edit-form.component";
+
+
+@Component({
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.scss']
+})
+
+export class DashboardComponent implements OnInit {
+  boardId = '';
+  columns: Array<any> = [];
+  userDataId = '';
+  @Output() updatedTaskDataEvent: EventEmitter<Task> = new EventEmitter();
+  selectedColumn:any;
+  boardTitle!: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private dashboardsService: DashboardService,
+    private dialog: MatDialog,
+    private location: Location,
+    private changeDetector: ChangeDetectorRef,
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe((params): void => {
```

Comment:
Every time visiting this page will be created a new subscription. This leads to performance issue. So you should pay attention to closing them. It might be unsubscribing ( in the onDestroy method) or approaches from the article https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f. In the projects' life, we use take* more often than others.